lib/index.ts

```diff
@@ -704,7 +704,9 @@ export class DockerProgress {
 		if (options?.authconfig) {
 			const { serveraddress, ...authconfig } = options.authconfig;
 			options.registryconfig = {
-				[serveraddress]: authconfig,
+				...((serveraddress && authconfig.username && authconfig.password
```

Comment:
What is the reason for this change? Did you see issues from misconfigured auth?

Reply:
Could there be another type of `authconfig` (e.g. token) that is also valid?

Reply:
It was to satisfy Dockerode type complaints. In [Dockerode types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/dockerode/index.d.ts#L1009-L1025), `AuthConfig` has optional `username` and `password` fields, but `RegistryConfig` has non-optional `username` and `password`. `AuthConfig` also does have other auth methods such as a token, but `RegistryConfig` only specifies username & password (in Dockerode types anyway)

Reply:
I don't think I entirely trust the Dockerode types, we've seen them be flaky in the past. If there is a type issue perhaps doing `authconfig as RegistryConfig` may solve the problem?

Reply:
Simplified to:
```
...(serveraddress && { [serveraddress]: authconfig } as RegistryConfig),
```

as `serveraddress` could also be undefined for `AuthConfig` but isn't for `RegistryConfig`